runtime/doc/lua.txt

```diff
@@ -3349,4 +3349,194 @@ totable({f}, {...})                                       *vim.iter.totable()*
+                • headers HeaderTable | table<string, string | string[]> | nil Headers to set on
+                  the request
+                • redirect string|nil Redirect mode. Defaults to "follow".
+                  Possible values are:
+                  • "follow": Follow all redirects incurred when fetching a
+                    resource.
+                  • "error": Throw an error using on_err or vim.notify when
+                    status is 3XX.
+
+                • data string|table|nil Data to send with the request. If a
+                  table, it will be JSON-encoded. vim.net does not currently
+                  support form encoding.
+                • on_complete fun(response: table)|nil Callback function when
+                  request is completed successfully. The response has the
+                  following keys:
+                  • ok boolean Whether the request was successful (status
+                    within 2XX range).
+                  • headers fun(): HeaderTable Function returning a HeaderTable of response headers.
+                  • text fun(): string|nil Function returning response body.
+                    If method was HEAD, this is nil.
+                  • json fun(opts: table|nil): table|nil Read the body as
+                    JSON. Optionally accepts opts from |vim.json.decode|. Will
+                    throw errors if body is not JSON-decodable. Nil if method
+                    is HEAD.
+                  • method string The http method used in the most recent HTTP
+                    request.
+                  • status number The numerical response code.
+                  • size number The total amount of bytes that were
+                    downloaded. This is the size of the body/data that was
+                    transferred, excluding headers.
+                  • http_version number HTTP version used in the request.
+
+                • on_err fun(err: string[])|nil Function recieving a
+                  `stderr_buffered` string[] of error. err is either curl
+                  stderr or internal fetch() error. Without providing this
+                  function, |vim.net.fetch()| will automatically raise the
+                  error to the user. See |on_stderr| and `stderr_buffered`.
+
+    Return: ~
+        (number) jobid A job id.
+
+    See also: ~
+      • |vim.net.download()|
+      • https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
+      • |job-control|
+      • man://curl
+
+HeaderTable:append({self}, {key}, {value})              *HeaderTable:append()*
+    Append value to header.
+
+    Parameters: ~
+      • {self}  
+      • {key}   (string) Non case-sensitive header name.
+
+HeaderTable:get({self}, {key})                             *HeaderTable:get()*
+    Get header values.
+
+    Parameters: ~
+      • {self}  
+      • {key}   (string) Non case-sensitive header name.
+
+    Return: ~
+        string[]
```

Comment:
majority of the headers have single value. I'd expect `:get()` to return the `string` like how it's actually transported in the http header. Most http client library's Headers interface does that. 🤔

Reply:
it is now `string[] | string | nil`

Reply:
That's a bad api design tho. Return type should be consistent, it shouldn't return `string` or `string[]`.

For example,

Python has a `get_all` that always returns `string[]` and `get` that always returns `string`.

Javascript has `get` that always returns `string` and `getSetCookie` always returns `string[]`.

Golang has `get` that always returns `string`.

Of course, they can also return `null` / `nil` / equivalent.